Use splice() instead of repeated shift() to remove from queue

TysonAndre opened the pull request:
Calling splice() is more efficient than shift()ing from the start of the array up to batchSize(default of 300) times.
The amount of time taken by shift() is proportional to the length of the array. This would only matter for large arrays.

seemk commented:
Thanks a lot, this is a good catch, but the commits need to be signed, else I can't merge :(

TysonAndre commented:
> Thanks a lot, this is a good catch, but the commits need to be signed, else I can't merge :(

Fixed and added a gpg key to my GitHub account.

I didn't see that in https://github.com/signalfx/signalfx-nodejs/blob/main/CONTRIBUTING.md#contributing-via-pull-requests - maybe that (and other signalfx repos, if needed) should link to https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-gpg-key as a
